src/borg/testsuite/compress.py

```diff
@@ -116,40 +116,53 @@ def test_auto():
 
 
 @pytest.mark.parametrize(
-    "specs, c_type, result_range, obfuscate_factor",
+    "specs, c_type, result_range, obfuscation_factor",
     [
         ("obfuscate,1,none", CNONE, 50, 101),
-        ("obfuscate,110,none", CNONE, None, 1024),
         ("obfuscate,2,lz4", LZ4, 10, 1001),
         ("obfuscate,6,zstd,3", ZSTD, 90, 10000001),
         ("obfuscate,2,auto,zstd,10", Auto, 10, 1001),
     ],
 )
-def test_obfuscate(specs, c_type, result_range, obfuscate_factor):
+def test_factor_obfuscation(specs, c_type, result_range, obfuscation_factor):
+    # Testing relative random reciprocal size variation, obfuscation spec 1 to 6 inclusive
+    # obfuscate_factor = 10**(obfuscation spec) +1
     cs = CompressionSpec(specs)
     assert isinstance(cs.inner.compressor, c_type)
     compressor = cs.compressor
     data = bytes(10000)
     _, compressed = compressor.compress({}, data)
-
-    # tests obfuscation with compression
-    if not isinstance(cs.inner.compressor, CNONE):
+    if c_type == CNONE:  # no compression
+        assert len(data) <= len(compressed) <= len(data) * obfuscation_factor
+    else:  # with compression
         min_compress, max_compress = 0.2, 0.001  # estimate compression factor outer boundaries
-        assert max_compress * len(data) <= len(compressed) <= min_compress * len(data) * obfuscate_factor
-        assert len({len(compressor.compress({}, data)[1]) for i in range(100)}) > result_range
-        # compressing 100 times the same data should give multiple different result sizes
-
-    # tests obfuscation without compression
-    elif specs == "obfuscate,1,none":
-        assert len(data) <= len(compressed) <= len(data) * obfuscate_factor
-        assert len({len(compressor.compress({}, data)[1]) for i in range(100)}) > result_range
-    elif specs == "obfuscate,110,none":
-        data = bytes(1000)
-        _, compressed = compressor.compress({}, data)
-        assert 1000 <= len(compressed) <= 1000 + obfuscate_factor
-        data = bytes(1100)
+        assert max_compress * len(data) <= len(compressed) <= min_compress * len(data) * obfuscation_factor
+    assert len({len(compressor.compress({}, data)[1]) for i in range(100)}) > result_range
+    # compressing 100 times the same data should give multiple different result sizes
+
+
+@pytest.mark.parametrize(
+    "specs, c_type, obfuscation_padding",
+    [
+        ("obfuscate,110,none", CNONE, 1024),  # up to 1KiB padding
+        ("obfuscate,120,lz4", LZ4, 1048576),  # up to 1MiB padding
+        ("obfuscate,123,zstd,3", ZSTD, 8388608),  # max, up to 8MiB padding
```

Comment:
maybe use `2**10`, `2**20`, `2**23` to make the relation to the obfuscation level more obvious?